### PR TITLE
TMG changes

### PR DESCRIPTION
**Pull request by james-appleby**
Applying the custom features added to the ACS commons library for TMG custom features

### CODE REVIEW

File: bundle/src/main/java/com/adobe/acs/commons/images/CropConstants.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.adobe.acs.commons.images;
```
Comment (mehtarachna) [inline bundle/src/main/java/com/adobe/acs/commons/images/CropConstants.java]: This class is already in telegraph-components

File: bundle/src/main/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImpl.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import org.slf4j.LoggerFactory;
+
+import com.adobe.acs.commons.images.CropConstants;
+import com.adobe.acs.commons.images.ImageTransformer;
+import com.day.image.Layer;
+
+@Component(label = "ACS AEM Commons - Image Transformer - Crop with URL Parameters")
+@Properties({@Property(name = ImageTransformer.PROP_TYPE, value = ParamCropImageTransformerImpl.TYPE, propertyPrivate = true)})
+@Service
+public class ParamCropImageTransformerImpl extends CropImageTransformerImpl {
```
Comment (mehtarachna) [inline bundle/src/main/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImpl.java]: +1, already in telegraph-components

File: bundle/src/test/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImplTest.java [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ParamCropImageTransformerImplTest {
```
Comment (mehtarachna) [inline bundle/src/test/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImplTest.java]: +1

**Review by mehtarachna — CHANGES_REQUESTED** [review]
I think we only need to add NamedImageTransformerServlet & it's unit test class in this acs commons package, rest of the classes are in core.  This work of separating out classes from core and acs commons (so that patch size is minimum with each upgrade) was done before 6.2 but then 6.2 reverted it back due to some unknown reasons.

File: bundle/src/main/java/com/adobe/acs/commons/images/CropConstants.java (hunk 1 above)
Comment (james-appleby) [reply bundle/src/main/java/com/adobe/acs/commons/images/CropConstants.java]: Removed

File: bundle/src/main/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImpl.java (hunk 2 above)
Comment (james-appleby) [reply bundle/src/main/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImpl.java]: Removed

File: bundle/src/test/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImplTest.java (hunk 3 above)
Comment (james-appleby) [reply bundle/src/test/java/com/adobe/acs/commons/images/transformers/impl/ParamCropImageTransformerImplTest.java]: Removed

**Review by mehtarachna — APPROVED** [review]

**Review by OlutayoA — APPROVED** [review]

**Pull request merged**